packages/helpers/src/category/getCategoryUrl.ts

```diff
@@ -45,3 +45,27 @@ export function getCategoryUrl(category: LinkedCategory): string {
         : `/${getEntityPrefix(category)}/${category.id}`;
   }
 }
+
+// TODO: move to separate file and add tests
+export function getCategoryRoute(category: LinkedCategory) {
+  return {
+    path: getCategoryUrl(category),
+    state: {
+      routeName: "frontend.navigation.page",
+      foreignKey: category?.id,
+    },
+  };
+}
+
+// TODO: move to separate file and add tests
+export function getNavigationRoute(navigationElement: LinkedCategory) {
+  if (navigationElement.linkType == "product") {
```

Comment:
This is a specific navigation element. You can see that in menu position CMS -> Product with tier prices.
![image](https://user-images.githubusercontent.com/13100280/236523712-97661fb3-a56f-40aa-a6f0-a092186bcc10.png)

it links directly to the product page, not to the category page like any other menu link
@niklaswolf did you encounter such case?

Reply:
No :D didn't even know that this is possible.
I assume there is no reference to the `productId` in the `navigationElement`? If there were, we of course could easily provide the information to the state object.

Reply:
There is a `type` of navigation element. It's now as it should :)